### PR TITLE
fix: encode x-apigateway-event and x-apigateway-context; fixes #60

### LINKED ISSUE #60: The header content contains invalid characters
When running local test with the following `api-gateway-event.json`:

```json
{
  "httpMethod": "GET",
  "//body": "{\"name\": \"Jake\"}",
  "path": "/users",
  "resource": "/{proxy+}",
  "queryStringParameters": {
    "foo": "💩"
  },
  "pathParameters": {},
  "headers": {},
  "requestContext": {}
}
```

the following error occurs:

```
TypeError: The header content contains invalid characters
    at ClientRequest.OutgoingMessage.setHeader (_http_outgoing.js:358:11)
    at new ClientRequest (_http_client.js:86:14)
    at Object.exports.request (http.js:31:10)
    at forwardRequestToNodeServer (/Users/johnDoe/Developer/github.com/aws-serverless-express/example/node_modules/aws-serverless-express/index.js:115:22)
    at Server.startServer.on.error (/Users/johnDoe/Developer/github.com/aws-serverless-express/example/node_modules/aws-serverless-express/index.js:165:21)
    at emitNone (events.js:91:20)
    at Server.emit (events.js:185:7)
    at emitListeningNT (net.js:1285:10)
    at _combinedTickCallback (internal/process/next_tick.js:71:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

Any idea how to avoid this? Basically anyone adding weird character (emoji) to a query parameter can now crash the app.

### PR DESCRIPTION
**Pull request by brettstack**

### CODE REVIEW

File: __tests__/middleware.js [hunk 1]
```diff
@@ -5,8 +5,13 @@ const mockNext = () => true
 const generateMockReq = () => {
     return {
         headers: {
-            'x-apigateway-event': '{"path": "/foo/bar"}',
-            'x-apigateway-context': '{"foo": "bar"}'
+            'x-apigateway-event': encodeURIComponent(JSON.stringify({
+              path: '/foo/bar',
+              queryStringParameters: {
+                  foo: '💩'
```
Comment (kahouieong) [inline __tests__/middleware.js]: It might not be a good emoji in the repo. 😅

File: __tests__/middleware.js [hunk 2]
```diff
@@ -5,8 +5,13 @@ const mockNext = () => true
...
+              }
+            })),
+            'x-apigateway-context': encodeURIComponent(JSON.stringify({foo: 'bar'}))
```
Comment (kahouieong) [inline __tests__/middleware.js]: Do we want to encode those characters !'()* as well?

Comment (brettstack) [inline __tests__/middleware.js]: They get encoded correctly. Added to test.

File: __tests__/middleware.js (hunk 1 above)
Comment (brettstack) [inline __tests__/middleware.js]: Replaced 🖖

**Pull request merged**